File: lib/entity/session_entity.dart
import 'package:tencent_im_plugin/entity/group_info_entity.dart';
import 'package:tencent_im_plugin/entity/message_entity.dart';
import 'package:tencent_im_plugin/entity/user_info_entity.dart';
import 'package:tencent_im_plugin/utils/enum_util.dart';

/// 会话实体
class SessionEntity {
  // 封面(头像)
  String faceUrl;

  // 未读消息数量
  int unreadMessageNum;

  // 昵称(用户:用户昵称，群:群昵称，系统:空)
  String nickname;

  // 会话ID(C2C：对方账号；Group：群组Id)
  String id;

  // 会话类型
  SessionType type;

  // 最近一条消息
  MessageEntity message;

  // 群
  GroupInfoEntity group;

  // 用户
  UserInfoEntity userProfile;

  SessionEntity({
    this.faceUrl,
    this.unreadMessageNum,
    this.nickname,
    this.id,
    this.type,
    this.message,
    this.group,
    this.userProfile,
  });

  SessionEntity.fromJson(Map<String, dynamic> json) {
    faceUrl = json['faceUrl'];
    unreadMessageNum = json['unreadMessageNum'];
    nickname = json['nickname'];
    id = json['id'];
    for (var item in SessionType.values) {
      if (EnumUtil.getEnumName(item) == json['type']) {
        type = item;
      }
    }
    message = json["message"] != null
        ? MessageEntity.fromJson(json["message"])
        : null;

    group =
        json["group"] != null ? GroupInfoEntity.fromJson(json["group"]) : null;
    userProfile = json["userProfile"] != null
        ? UserInfoEntity.fromJson(json["userProfile"])
        : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['faceUrl'] = this.faceUrl;
    data['unreadMessageNum'] = this.unreadMessageNum;
    data['nickname'] = this.nickname;
    data['id'] = this.id;
    data['type'] = this.type.toString();
    data['message'] = this.message.toJson();
    data['group'] = this.group.toJson();
    data['userProfile'] = this.userProfile.toJson();
    return data;
  }

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is SessionEntity &&
          runtimeType == other.runtimeType &&
          id == other.id &&
          type == other.type;

  @override
  int get hashCode => id.hashCode ^ type.hashCode;
}

/// 会话类型
enum SessionType {
  // 用户对话
  C2C,
  // 系统对话
  System,
  // 群对话
  Group,
}

File: ios/Classes/entity/SessionEntity.swift
import ImSDK

//  Created by 蒋具宏 on 2020/2/10.
//  会话对象
public class SessionEntity : NSObject{
    /**
     * 会话 ID
     */
    var id : String?;
    
    /**
     * 会话名，如果是用户则为用户昵称，如果是群则为群昵称
     */
    var nickname : String?;
    
    /**
     * 会话类型
     */
    var type : SessionType?;
    
    /**
     * 头像(用户头像、群头像、系统头像)
     */
    var faceUrl : String?;
    
    /**
     * 未读消息数量
     */
    var unreadMessageNum : Int32?;
    
    /**
     * 最近一条消息
     */
    var message : MessageEntity?;
    
    /**
     * 群信息，在type为群时有效
     */
    var group : GroupInfoEntity?;
    
    /**
     * 用户信息，在type为C2C时有效
     */
    var userProfile : UserInfoEntity?;
}
